resources/2024-06-support-secretstores-env.md

```diff
@@ -0,0 +1,417 @@
+model SecretReference {
+  @doc("The ID of an Applications.Core/SecretStore resource containing sensitive data required for recipe execution.")
+  source: string;
+
+  @doc("The key for the secret in the secret store.")
+  key: string;
+}
+```
+
+
+
+
+### Implementation Details
+<!--
+High level description of updates to each component. Provide information on 
+the specific sub-components that will be updated, for example, controller, processor, renderer,
+recipe engine, driver, to name a few.
+-->
+The renderer will need to be updated in several areas to handle the new secrets implementation.
+
+The function **GetDependencyIDs** will need to be updated to handle the new secret reference type. This function will need to determine if the environment variable is a secret reference or a string. The function will also need to determine whether the secret is a radius resource or a Kubernetes secret.
+
+The function **convertEnvVar** will need to be created to facilitate the conversion of `map[string]EnvironmentVariable` to `map[string]corev1.EnvVar`. The function will need to handle resolving the secret coming from a Kubernetes secret or a Radius resource ID.
+
+
+#### Core RP (if applicable)
+
+### Error Handling
+<!--
+Describe the error scenarios that may occur and the corresponding recovery/error handling and user experience.
+-->
+Error handling is covered within the functions and Radius errors are used where appropriate.
+
+## Test plan
+
+<!--
+Include the test plan to validate the features including the areas that
+need functional tests.
+
+Describe any functionality that will create new testing challenges:
+- New dependencies
+- External assets that tests need to access
+- Features that do I/O or change OS state and are thus hard to unit test
+-->
+### Unit Tests
+- The addition of tests for the conversion functions from and to the versioned datamodel to the base datamodel.
+- Update functional tests to cover the new functionality.
+
+## Security
+
+<!--
+Describe any changes to the existing security model of Radius or security 
+challenges of the features. For each challenge describe the security threat 
+and its mitigation with this design. 
+
+Examples include:
+- Authentication 
+- Storing secrets and credentials
+- Using cryptography
+
+If this feature has no new challenges or changes to the security model
+then describe how the feature will use existing security features of Radius.
+-->
+The handling of secrets will remain within Kubernetes and Radius is only providing a way to reference these secrets in the environment variables of a container. This is an improvement over the current method of mounting secrets as volumes as it allows for more flexibility and security. Also, the secrets are stored in the Kubernetes secret store and are never exposed to the user.
+
+## Compatibility (optional)
```

Comment:
given the breaking nature of the proposed changes, we should amplify this design to the community to get as many eyes on this review as possible

Reply:
Created a post in discord to notify the larger community - https://discord.com/channels/1113519723347456110/1277988146834837529